### PR TITLE
[FIX] account: fix list view edit in studio

### PR DESCRIPTION
**Pull request by adda-odoo**
Issue -->

Traceback when clicking `Edit List View` under Invoice Lines in the `account.move` form view while using Studio.
The cause of the issue is that the props update and lose the data that we expect in the getter `isProductClickable`.

Solution -->

Use an alternate way to get the parent record (account.move) in the case that the root model is not account.move.

opw-4100711

### CODE REVIEW

File: addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js [hunk 1]
```diff
@@ -162,7 +162,12 @@ export class ProductLabelSectionAndNoteField extends Many2OneField {
     }
 
     get isProductClickable() {
-        return this.props.record.model.root.data.state !== "draft";
+        if (this.props.record.model.root.data) {
+            return this.props.record.model.root.data.state !== "draft";
+        }
+        else {
+            return this.props.record._parentRecord.data.state !== "draft";
+        }
```
Comment (rugo-odoo) [inline addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js:170]:
```suggestion
        if (this.props.record.model.root.data) {
            return this.props.record.model.root.data.state !== "draft";
        }
            
        return this.props.record._parentRecord.data.state !== "draft";
```
No need for the else, we can just use a guard clause.

**Review by rugo-odoo — CHANGES_REQUESTED** [review]

**Review by Megaaaaaa — APPROVED** [review]
Thank you for the fix, LGTM !

**Comment by oco-odoo** [comment]
@odoo/rd-framework-js  Could one of you confirm this is ok ? It functionally fixes the problem, but I'm not too sure about the deep JS framework logic ^^

File: addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js [hunk 2]
```diff
@@ -162,7 +162,11 @@ export class ProductLabelSectionAndNoteField extends Many2OneField {
     }
 
     get isProductClickable() {
-        return this.props.record.model.root.data.state !== "draft";
+        if (this.props.record.model.root.data) {
+            return this.props.record.model.root.data.state !== "draft";
+        }
+        
+        return this.props.record._parentRecord.data.state !== "draft";
```
Comment (aab-odoo) [inline addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js:169]: General remark: you're not supposed to access private properties (prefixed by _). I don't know what the issue/fix are though. If it's related to studio, @kebeclibre would know

**Pull request closed without merging**